pyccel/codegen/printing/fcode.py

```diff
@@ -2311,7 +2311,8 @@ def _print_Allocate(self, expr):
             elif expr.alloc_type == 'reserve':
                 var_code = self._print(expr.variable)
                 size_code = self._print(expr.shape[0])
-                return f'call {var_code} % reserve({size_code})\n'
+                return (f'call {var_code} % clear()\n'
+                        f'call {var_code} % reserve({size_code})\n')
```

Comment:
Can the `clear` call be avoided if the status is `unallocated`?

Reply:
good catch!